### PR TITLE
Containers: Add VS Code tasks and improve overall documentation & flexibility 

### PR DESCRIPTION
**Pull request by bpmct**
Thanks to @AurangAsif for the feedback on this and creating the VS Code tasks 🎉 see #8 

- START_DIR can now be specified
- add VS Code tasks for rclone!
- document HASHED_PASSWORD
- fix install extension instructions in Dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,15 +13,18 @@ ENV SHELL=/bin/bash
 RUN sudo apt-get update && sudo apt-get install unzip -y
 RUN curl https://rclone.org/install.sh | sudo bash
 
+# Copy rclone tasks to /tmp, to potentially be used
+COPY deploy-container/rclone-tasks.json /tmp/rclone-tasks.json
+
+# Fix permissions for code-server
+RUN sudo chown -R coder:coder /home/coder/.local
+
```
Comment (bpmct) [inline Dockerfile:21]: this makes the install extension example work

File: deploy-container/README.md [hunk 2]
```diff
@@ -13,12 +13,13 @@ Docker Hub: `bencdr/code-server-deploy-container`
...
-| Variable Name | Description                                                | Default Value |
-| ------------- | ---------------------------------------------------------- | ------------- |
-| `PASSWORD`    | Password for code-server                                   |
-| `USE_LINK`    | Use code-server --link instead of a password (coming soon) | false         |
-| `GIT_REPO`    | A git repository to clone                                  |               |
-
+| Variable Name     | Description                                                                                      | Default Value       |
+| ----------------- | ------------------------------------------------------------------------------------------------ | ------------------- |
+| `PASSWORD`        | Password for code-server                                                                         |                     |
+| `HASHED_PASSWORD` | Overrrides PASSWORD. [SHA-256 hash](https://xorbin.com/tools/sha256-hash-calculator) of password |
```
Comment (bpmct) [inline deploy-container/README.md:19]: added hashed password :)

File: deploy-container/README.md [hunk 3]
```diff
@@ -44,18 +45,21 @@ Now, you can add the following the environment variables in the code-server clou
...
-# How to use:
+# --- How to use ---
 
+# Terminal:
 $ sh /home/coder/push_remote.sh # save your uncomitted files to the remote
-
 $ sh /home/coder/pull_remote.sh # get latest files from the remote
+
+# In VS Code:
+# ctrl + P, run task: push_remote or pull_remote
```
Comment (bpmct) [inline deploy-container/README.md:62]: documented the new tasks

File: deploy-container/rclone-tasks.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      "label": "push_remote",
+      "type": "shell",
+      "command": "sh /home/coder/push_remote.sh",
+      "presentation": {
+        "reveal": "always"
+      },
+      "problemMatcher": [],
+      "options": {
+        "statusbar": {
+          "label": "$(repo-push) rclone: push"
```
Comment (bpmct) [inline deploy-container/rclone-tasks.json:14]: i slightly modified the labels

**Review by bpmct — COMMENTED** [review]
@AurangAsif would love your thoughts

**Comment by bpmct** [comment]
> It would also be useful if the user could disable the automatic pull from rclone at startup.

I can disable this but what is the use case? It will only pull if there are existing files in the rclone remote. Should I also make a setting so that it doesn't push the files by default?

**Comment by AurangAsif** [comment]
I've noticed pulling from rclone after starting code-server is much faster. Not sure why but it may just be my case.
Having the option to not push automatically would be nice but it's also fine without it. One thing to note is maybe documenting how to change the push and pull remote .sh files to pass in flags to rclone sync. I have been using the exclude flag to stop some files from being pushed and its really handy.

**Comment by bpmct** [comment]
Just added the flags and the disable push/pull vars and documented. Thanks for the help!

**Pull request merged**